File: src/test/java/com/fareFinder/AirportTest.java
package com.fareFinder;

import org.junit.Test;
import static org.junit.Assert.*;

public class AirportTest {

    @Test
    public void testKnownAirports() {
        assertEquals("SFO", Airport.lookup("San Francisco", "CA"));
        assertEquals("JFK", Airport.lookup("New York", "NY"));
        assertEquals("ORD", Airport.lookup("Chicago", "IL"));
        assertEquals("ATL", Airport.lookup("Atlanta", "GA"));
        assertEquals("DEN", Airport.lookup("Denver", "CO"));
        assertEquals("SEA", Airport.lookup("Seattle", "WA"));
        assertEquals("MIA", Airport.lookup("Miami", "FL"));
        assertEquals("LAS", Airport.lookup("Las Vegas", "NV"));
        assertEquals("BOS", Airport.lookup("Boston", "MA"));
        assertEquals("DFW", Airport.lookup("Dallas", "TX"));
    }

    @Test(expected = IllegalArgumentException.class)
    public void testUnknownCityThrows() {
        Airport.lookup("Nonexistent City", "XX");
    }
}

File: go/tests/airport_test.go
package tests

import (
	"testing"

	fare_finder "github.com/lakamsani/fare-finder/fare_finder"
)

func TestKnownAirports(t *testing.T) {
	cases := []struct {
		city, state, want string
	}{
		{"San Francisco", "CA", "SFO"},
		{"New York", "NY", "JFK"},
		{"Chicago", "IL", "ORD"},
		{"Atlanta", "GA", "ATL"},
		{"Denver", "CO", "DEN"},
		{"Seattle", "WA", "SEA"},
		{"Miami", "FL", "MIA"},
		{"Las Vegas", "NV", "LAS"},
		{"Boston", "MA", "BOS"},
		{"Dallas", "TX", "DFW"},
	}

	for _, tc := range cases {
		got, err := fare_finder.LookupAirport(tc.city, tc.state)
		if err != nil {
			t.Errorf("LookupAirport(%q, %q): unexpected error: %v", tc.city, tc.state, err)
			continue
		}
		if got != tc.want {
			t.Errorf("LookupAirport(%q, %q) = %q, want %q", tc.city, tc.state, got, tc.want)
		}
	}
}

func TestUnknownCityReturnsError(t *testing.T) {
	_, err := fare_finder.LookupAirport("Nonexistent City", "XX")
	if err == nil {
		t.Fatal("expected error for unknown city, got nil")
	}
}
